# Hi, I am Aashish and i am going to explain about Multiple regression.

## Firstly i used pandas library to import csv file.

In [3]:
import pandas as pd
df = pd.read_csv("Stock_data.csv")

In [5]:
print(df)

     Stock  Volatility  Market Return  Stock return
0       S1        0.15          0.052       0.11460
1       S2        0.20          0.002       0.04520
2       S3        0.18          0.094       0.17376
3       S4        0.69          0.087       0.17008
4       S5        0.19          0.013       0.06048
..     ...         ...            ...           ...
995   S996        0.46          0.043       0.10572
996   S997        0.18          0.087       0.16396
997   S998        0.68          0.031       0.09156
998   S999        0.48          0.040       0.10176
999  S1000        0.61          0.019       0.07392

[1000 rows x 4 columns]


## Then make a list of the independent values and call this variable x and then put the dependent values in a variable called y.

In [8]:
x = df[['Volatility', 'Market Return']]
y = df['Stock return']

## I will use some methods from the sklearn module and from the sklearn module we will use the LinearRegression() method to create a linear regression. Then i will fit the independent and dependent values by using fit()

In [11]:
from sklearn import linear_model

regr = linear_model.LinearRegression()
regr.fit(x, y)

LinearRegression()

## Calculating coefficients

In [13]:
print(regr.coef_)

[0.012 1.4  ]


## Calculating Intercept

In [15]:
print(regr.intercept_)

0.039999999999999966


## Predicting stock return when by asumming market return is 0.08 and volatility is 0.31.

In [17]:
prediction = regr.predict([[0.31, 0.08]])

print(prediction)

[0.15572]


C:\Users\praveen\anaconda3\Lib\site-packages\sklearn\base.py:493: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


# Therefore possible linear equation is : Stock return = 0.012*(Volatility) + 1.2*(Market return) + 0.04